File: src/game/utils/Paths.java
package game.utils;

import game.Bomberman;
import game.level.Level;

public class Paths {
    public static String getPlayerDataPath() {
        return String.format("%s/data", Paths.getDataFolder());
    }

    public static String getDataFolder() {
        return "data";
    }

    public static String getAssetsFolder() {
        return "assets";
    }

    public static String getEntitiesFolder() {
        return String.format("%s/entities", getAssetsFolder());
    }

    public static String getEnemiesFolder() {
        return String.format("%s/enemies", getEntitiesFolder());
    }

    public static String getWorldsFolder() {
        return String.format("%s/worlds", getAssetsFolder());
    }

    public static String getCurrentLevelFolder() {
        Level level = Bomberman.getMatch().getCurrentLevel();
        return String.format("%s/%d/level/%d", getWorldsFolder(), level.getWorldId(), level.getLevelId());
    }

    public static String getCurrentWorldCommonFolder() {
        return String.format("%s/%d/common", getWorldsFolder(), Bomberman.getMatch().getCurrentLevel().getWorldId());
    }

    public static String getPowerUpsFolder() {
        return String.format("%s/powerups", getAssetsFolder());
    }

    public static String getPowerUpsBorderPath() {
        return String.format("%s/powerups_border.png", getMenuImagesPath());
    }

    public static String getMenuImagesPath() {
        return String.format("%s/menu/images", getAssetsFolder());
    }

    public static String getBackgroundImage() {
        return String.format("%s/background.jpg", getMenuImagesPath());
    }

    public static String getMainMenuWallpaper() {
        return String.format("%s/welcome.png", getMenuImagesPath());
    }

    public static String getDeathWallpaper() {
        return String.format("%s/death.jpg", getMenuImagesPath());
    }

    public static String getWorldSelectorPortalPath(int id) {
        return String.format("%s/world_selector/world%d.png", Paths.getAssetsFolder(), id);
    }

    public static String getPowerupsLogoPath() {
        return String.format("%s/powerups_logo.png", getPowerUpsFolder());
    }

    public static String getInventoryPath() {
        return String.format("%s/inventory", getAssetsFolder());
    }

    public static String getSoundsPath() {
        return String.format("%s/sounds", getAssetsFolder());
    }

    public static String getItemsPath() {
        return String.format("%s/items", getAssetsFolder());
    }

    public static String getDefaultSoundTrack() {
        return String.format("%s/menu/sound/soundtrack.wav", getAssetsFolder());
    }

    public static String getUiFolder() {
        return String.format("%s/ui", getAssetsFolder());
    }


    public static String getCursorPath() {
        return String.format("%s/cursor.png", getUiFolder());
    }

    public static String getIconPath() {
        return getMenuImagesPath() + "/frame_icon.png";
    }
}

File: src/game/utils/Paths.kt
package game.utils

import game.Bomberman

object Paths {
    const val dataFolder: String = "data"
    private const val assetsFolder: String = "assets"
    const val playerDataPath: String = "$dataFolder/data"
    const val entitiesFolder: String = "$assetsFolder/entities"
    const val enemiesFolder: String = "$entitiesFolder/enemies"
    const val worldsFolder: String = "$assetsFolder/worlds"
    val currentLevelFolder: String get() = Bomberman.getMatch().currentLevel.let { "$worldsFolder/${it.info.worldId}/level/${it.info.levelId}" }
    val currentWorldCommonFolder: String get() = String.format("%s/%d/common", worldsFolder, Bomberman.getMatch().currentLevel.info.worldId)
    const val powerUpsFolder: String = "$assetsFolder/powerups"
    const val menuImagesPath: String = "$assetsFolder/menu/images"
    const val powerUpsBorderPath: String = "$menuImagesPath/powerups_border.png"
    const val backgroundImage: String = "$menuImagesPath/background.jpg"
    const val mainMenuWallpaper: String = "$menuImagesPath/welcome.png"
    const val deathWallpaper: String = "$menuImagesPath/death.jpg"
    fun getWorldSelectorPortalPath(id: Int): String = "$assetsFolder/world_selector/world$id.png"
    const val powerupsLogoPath: String = "$powerUpsFolder/powerups_logo.png"
    const val inventoryPath: String = "$assetsFolder/inventory"
    const val soundsPath: String = "$assetsFolder/sounds"
    const val itemsPath: String = "$assetsFolder/items"
    const val defaultSoundTrack: String = "$assetsFolder/menu/sound/soundtrack.wav"
    const val uiFolder: String = "$assetsFolder/ui"
    const val cursorPath: String = "$uiFolder/cursor.png"
    const val iconPath: String = "$menuImagesPath/frame_icon.png"
    const val xmlPath: String = "$assetsFolder/xml"
    const val skinsXml: String = "$xmlPath/skins.xml"
}